astropy/wcs/tests/test_utils.py

```diff
@@ -589,7 +589,9 @@ def test_noncelestial_scale(cdelt, pc, cd):
 
     # TODO: Some inputs emit RuntimeWarning from here onwards.
     #       Fix the test data. See @nden's comment in PR 9010.
-    mywcs.wcs.cdelt = cdelt
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', 'cdelt will be ignored since cd is present', RuntimeWarning)
+        mywcs.wcs.cdelt = cdelt
```

Comment:
Instead of always assigning to `CDELT` is it better to assign only when a `PC` is assigned and not when a `CD` is assigned? Technically it's incorrect to assign to `CDELT` when the `CD` is present. In that way a warning will not be emitted.

Reply:
The `TODO` comment should be removed when this is fixed.